### PR TITLE
Update excel.py to output OrderedDict instead of dict

### LINKED ISSUE #9930: read_excel with sheetname=None or sheetname=[0,1,”Sheet5”] should preserve sheets order
Hello,

I noticed in http://pandas.pydata.org/pandas-docs/stable/generated/pandas.read_excel.html#pandas.read_excel
that `read_excel` can have a parameter `sheetname`set to `None` or a to a `list`

In such a case a dictionary of `DataFrame` is returned.

Maybe it should output an `OrderedDict` of `DataFrame` to preserve sheets order.

Kind regards

### PR DESCRIPTION
**Pull request by s-celles**
It should fix https://github.com/pydata/pandas/issues/9930

### CODE REVIEW

File: pandas/io/excel.py [hunk 1]
```diff
@@ -8,6 +8,7 @@
 import datetime
 import abc
 import numpy as np
+from collections import OrderedDict
```
Comment (shoyer) [inline pandas/io/excel.py]: This should be imported from `pandas.compat` instead to ensure backwards compatibility with Python 2.6.

**Comment by shoyer** [comment]
This does seem like a good idea. Can you write a test to verify that it works?

**Comment by scls19fr** [comment]
Sorry but I don't understand why CI raises this error for Python 2.x

    Error: HTTPError: 403 Client Error: Forbidden: http://conda.binstar.org/pandas/linux-64/bigquery-2.0.17-py27_0.tar.bz2

I also don't understand why "OrderedDict should be imported from `pandas.compat` instead to ensure backwards" because according https://docs.python.org/2/library/collections.html collections exists since Python 2.4 "New in version 2.4." but maybe collections is existing since 2.4 but not OrderedDict.

I noticed that a pip package exists https://pypi.python.org/pypi/ordereddict but I don't know how to automatically install it only for Python < 2.7 (so 2.6 in our case)

**Comment by shoyer** [comment]
I think that CI failure is something spurious related to binstar. I just retriggered the Travis tests for this PR -- hopefully we won't see that error message again.

As for OrderedDict, indeed it first appeared in Python 2.7. We already have a copy of OrderedDict bundled with pandas for exactly this reason. To use it, just use `from pandas.compat import OrderedDict`.

**Comment by jorisvandenbossche** [comment]
Follow-up: #9944

**Pull request closed without merging**